### ISSUE TITLE
Move "react" to be a peer dependency

### ISSUE THREAD

**Issue by ide** [facebook/react-native#5541]
By moving "react" out of "dependencies" and into "peerDependencies" this unlocks the next level of universal code sharing. Packages like react-redux can have a peer dependency on the same copy of "react" that's used by RN. Short of abusing npm 3's package flattening, I believe this is the proper way forward.

Example package.json:
```json
{
  "dependencies": {
    "react": "^0.14.0",
    "react-native": "^0.20.0",
    "react-redux": "^4.0.0"
  }
}
```

... and that's it. This would be a breaking change, but the fix is very simple (add "react" to the app's dependencies). With the fbjs patches landing, this will be hopefully the final step to compatibility with React Redux, Relay, and the greater React ecosystem.

**Comment by mangogogos** [comment]
Definite +1!

**Comment by ide** [comment]
This is going to require a bit of work. These are the steps:

- Move "react" to peerDeps
- Make react-native-cli add "react" to the dependencies in the package.json file it generates. **This is a breaking change to the CLI but I think we have to do it.** cc @vjeux
  - Side note: react-native-cli should check for updates to itself and let you know if you need to `npm install -g` again.
- Figure out how to run the example projects like UIExplorer. I think the correct but annoying solution is to give each example project its own package.json file that depends on react and react-native.

**Comment by vjeux** [comment]
react-native init

Does
npm install react-native
./node_modules/react-native/cli.js install

So we can run arbitrary script that's going to add a peer dependency after the fact and re-run npm install without changing the shell

**Comment by ide** [comment]
Nice, that makes things a lot smoother...

**Comment by vjeux** [comment]
Yeah, @frantic did a good job designing the global cli such that it is future proof :)

**Comment by Proyoyo** [comment]
May I know when this fix will probably be released?

**Comment by frantic** [comment]
@Proyoyo see https://github.com/facebook/react-native/pull/5813

**Comment by janicduplessis** [comment]
#5813 landed in will be released in 0.22

**Comment by janicduplessis** [comment]
@facebook-github-bot close
